galaxy_ng/contrib/metrics_middleware.py

```diff
@@ -0,0 +1,50 @@
+import base64
+import json
+
+from django_prometheus.middleware import (
+    Metrics,
+    PrometheusAfterMiddleware,
+    PrometheusBeforeMiddleware,
+)
+
+from galaxy_ng.app.auth.auth import RHIdentityAuthentication
+
+
+EXTENDED_METRICS = [
+    "django_http_requests_latency_seconds_by_view_method",
+    "django_http_responses_total_by_status_view_method",
+    "django_http_requests_total_by_view_transport_method",
+]
+
+
+class AccountEnhancedMetrics(Metrics):
+    def register_metric(self, metric_cls, name, documentation, labelnames=(), **kwargs):
+        if name in EXTENDED_METRICS:
+            labelnames += ("account",)
+        return super().register_metric(
+            metric_cls, name, documentation, labelnames=labelnames, **kwargs
+        )
+
+
+class AccountEnhancedMetricsBeforeMiddleware(PrometheusBeforeMiddleware):
+    metrics_cls = AccountEnhancedMetrics
+
+
+class AccountEnhancedMetricsAfterMiddleware(PrometheusAfterMiddleware):
+    metrics_cls = AccountEnhancedMetrics
+
+    def label_metric(self, metric, request, response=None, **labels):
+        new_labels = labels
+        if metric._name in EXTENDED_METRICS:
+            account = "unknown"
+            encoded_identity = request.META.get(RHIdentityAuthentication.header)
```

Comment:
Seems like this will handle standalone cases where RHIdentityAuthentication will never be available by setting account="unknown".

So wondering if this should be toggled on/off based on DEPLOYMENT_MODE. But I also like using
ACCOUNT_ENHANCED_METRICS as the flag since it's not coupled to the deployment mode.
So as is is fine I think

Reply:
A new metric will be created/updated for every request?  

Instead of extracting it from the RHIdentity header each time, could set any of that info needed on the request once and then reference it here?  For ex, I think the request has the 'user' and the 'user' groups are essential the account number.

That would also provide an alternate 'account' value for standalone deployments.

Reply:
In a standalone case you wouldn't enable ACCOUNT_ENHANCED_METRICS so the `account="unknown"` wouldn't be there, the unknown would only be hit for routes like `/metrics` or a status endpoint that aren't hit by 3scale.